projects/packages/analyzer/src/class-utils.php

```diff
@@ -31,24 +31,37 @@ static function get_param_default_as_string( $default, $current_class ) {
 		}
 	}
 
-	static function node_to_class_name( $node, $class_for_self = null ) {
+	/**
+	 * Gets string representation of the Node's class
+	 *
+	 * @param Node $node Node.
+	 * @param any  $class_for_self Class reference.
+	 * @return string
+	 */
+	public static function node_to_class_name( $node, $class_for_self = null ) {
 		if ( $node instanceof Node\Expr\Variable
 			|| $node instanceof Node\Stmt\Class_ ) {
 			$class_name = $node->name;
 		} elseif ( $node instanceof Node\Name ) {
 			$class_name = '\\' . implode( '\\', $node->parts );
 		} elseif ( $node instanceof Node\Expr\PropertyFetch ) {
 			$class_name =
-						'$'
-						. self::maybe_stringify( $node->var->name )
-						. '->' . self::maybe_stringify( $node->name->name );
+				'$'
+				. self::maybe_stringify( $node->var->name )
+				. '->' . self::maybe_stringify( $node->name->name );
 		} elseif ( $node instanceof Node\Expr\ArrayDimFetch ) {
 
-			$class_name =
-						'$'
-						. self::maybe_stringify( $node->var->name )
-						. '[' . self::maybe_stringify( $node->dim->value )
-						. ']';
+			$dim_val = '';
+			if ( $node->dim instanceof Node\Expr\Variable ) {
+				$dim_val = '[$' . self::maybe_stringify( $node->dim->name ) . ']';
+			} elseif ( $node->dim instanceof Node\Expr\PropertyFetch ) {
+				$dim_val = '[$' . self::maybe_stringify( $node->dim->var->name ) . ']';
```

Comment:
Is this right? It looks like it should be something more like what's going on in lines 49–51.

For that matter, maybe there should just be a recursive call here (and in some of the other cases)?

Reply:
Indeed, it wasn't properly constructing the call. I updated similarly to 49-51

Reply:
As for refactoring it, I guess it makes sense. But I'd like to address it in further PRs